core/src/test/java/org/pentaho/platform/engine/services/runtime/RuntimeContextTest.java

```diff
@@ -0,0 +1,282 @@
+    params.put( "Two", new Long( 2 ) );
+    params.put( "Three", three );
+    params.put( "Date", now );
+
+    list.add( 1 );
+    list.add( 2 );
+    list.add( 3 );
+    params.put( "List", list );
+
+    final Collection readAttrs = null;
+    final String instanceId = runtimeCtx.createNewInstance( persisted, params, forceImmediateWrite );
+    final IRuntimeElement elem = repository.loadElementById( instanceId, readAttrs );
+
+    Assert.assertNotNull( elem );
+    Assert.assertTrue( elem.getStringProperty( "One" ).equals( "One" ) );
+    Assert.assertTrue( elem.getLongProperty( "Two", 0 ) == 2 );
+    Assert.assertTrue( elem.getBigDecimalProperty( "Three" ).equals( three ) );
+    Assert.assertTrue( elem.getDateProperty( "Date" ).equals( now ) );
+    Assert.assertNotNullOrEmpty( elem.getListProperty( "List" ) );
+  }
+
+  /**
+   * It seems that status and error level are one and alike by design.
+   * Enforce a test on this. In addition achieve code coverage on these.
+   *
+   * @throws Exception
+   */
+  @Test
+  public void testErrorLevel() throws Exception {
+    final RuntimeContext rctx = (RuntimeContext) runtimeCtx;
+    Assert.assertTrue( rctx.getErrorLevel() == rctx.getStatus() );
+  }
+
+  @Test
+  public void testPromptNow() throws Exception {
+    runtimeCtx.setPromptStatus( IRuntimeContext.PROMPT_NO );
+    Assert.assertFalse( runtimeCtx.isPromptPending() );
+
+    runtimeCtx.promptNow();
+    Assert.assertTrue( runtimeCtx.isPromptPending() );
+  }
+
+  @Test
+  public void testPromptNeeded() throws Exception {
+    runtimeCtx.setPromptStatus( IRuntimeContext.PROMPT_NO );
+    Assert.assertFalse( runtimeCtx.isPromptPending() );
+
+    runtimeCtx.promptNeeded();
+    Assert.assertTrue( runtimeCtx.isPromptPending() );
+  }
+
+  @Test
+  public void testGetUrlFactory() throws Exception {
+    Assert.assertNotNull( runtimeCtx.getUrlFactory() );
+    Assert.assertTrue( runtimeCtx.getUrlFactory() == urlFactory );
+  }
+
+  private Map<String, IParameterProvider> makeParameterProviders( final IPentahoSession session ) {
+    final Map<String, IParameterProvider> res = new HashMap<>();
+
+    res.put( "jvm", new JVMParameterProvider() );
+    res.put( SecurityParameterProvider.SCOPE_SECURITY, new SecurityParameterProvider( session ) );
+
+    return res;
+  }
+
+  private ISolutionEngine makeSolutionEngine( final IPentahoSession session ) {
+    final ISolutionEngine res = PentahoSystem.get( ISolutionEngine.class, session );
+    res.setLoggingLevel( ILogger.ERROR );
+    res.init( session );
+    return res;
+  }
+
+  private IRuntimeElement makeRuntimeData( final IPentahoSession session ) {
+    return new SimpleRuntimeElement( UUIDUtil.getUUIDAsString(), session.getId( ), IParameterProvider.SCOPE_SESSION );
+  }
+  /**
+   * Test oriented repository implementation, used for confirming expected results.
+   */
+  private static class RuntimeRepository extends PentahoBase implements IRuntimeRepository {
```

Comment:
It will require spending quite some time on learning mock and I am not sure I should spend the time on that now.